microsoft/testsuites/dpdk/dpdkutil.py

```diff
@@ -310,6 +310,9 @@ def initialize_node_resources(
     extra_nics: Union[List[NicInfo], None] = None,
 ) -> DpdkTestResources:
     _set_forced_source_by_distro(node, variables)
+    if pmd == "failsafe" and node.nics.is_mana_device_present():
+        raise SkippedException("Failsafe PMD test on MANA is not supported.")
```

Comment:
Raise UnsupportedOsException in tools, and catch it in test cases to skip.

Reply:
That's easy to make some test cases should be failed to be skipped. But the tool is in test suite folder, if you think it's ok, you can do like it.